packages/gatsby-plugin-manifest/src/gatsby-browser.js

```diff
@@ -0,0 +1,26 @@
+import { withPrefix as fallbackWithPrefix, withAssetPrefix } from "gatsby"
+
+const withPrefix = withAssetPrefix || fallbackWithPrefix
+
+exports.onRouteUpdate = function({ location }, pluginOptions) {
+  const { localize } = pluginOptions
+  const manifestFilename = getManifestForPathname(location.pathname, localize)
+
+  const manifestEl = document.head.querySelector(`link[rel="manifest"]`)
+  if (manifestEl) {
+    manifestEl.setAttribute(`href`, withPrefix(manifestFilename))
+  }
+}
+
+function getManifestForPathname(pathname, localizedApps) {
+  let suffix = ``
+  if (Array.isArray(localizedApps)) {
```

Comment:
with this if statement and the one below I'd recommend you put the happy path outside the if. So, negate both these and return early. it helps make the code more readable to keep your happy path out of if statements when possible imo.

Reply:
I wanted to avoid repeating the string `manifest.webmanifest` in the function, but I wrote it out using the "return early" style to compare:

```javascript
function getManifestForPathname(pathname, localizedApps) {
  const defaultFilename = `manifest.webmanifest`
  if (!Array.isArray(localizedApps)) {
    return defaultFilename
  }

  const appOptions = localizedApps.find(app =>
    RegExp(`^${app.start_url}.*`, `i`).test(pathname)
  )
  if (!appOptions) {
    return defaultFilename
  }
  
  const suffix = `_${appOptions.lang}`
  return `manifest${suffix}.webmanifest`
}
```

I slightly prefer the current implementation because it avoids repeating the filename, but let me know if you'd like me to switch because I don't feel strongly about it. 😎

Reply:
I know it's a little more verbose, but I think it's more readable. :man_shrugging:

Reply:
Done!